File: Zig/Practice/Pointers/pass_by_ref.zig
const std = @import("std");
const print = std.debug.print;

pub fn main() void {
    var x: u8 = 10;
    print("Before passByRef: {}\n", .{x});
    passByRef(&x);
    print("Outside passByRef: {}\n", .{x});
}

fn passByRef(num: *u8) void {
    num.* = 20;
    print("Inside passByRef: {}\n", .{num.*});
}

File: languages/c/practice/pointers/pass_by_ref.c
#include <stdio.h>

void pass_by_ref(int *num);

int main(void)
{
    int x = 10;
    print("Before pass_by_ref: %d\n", x);
    pass_by_ref(&x);
    printf("Outside pass_by_ref: %d", x);
    return 0;
}

void pass_by_ref(int *num)
{
    *num = 20; // This change will affect the original variable
    printf("Inside pass_by_ref: %d\n", *num);
}